flow/compositor_context.cc

```diff
@@ -85,9 +118,16 @@ RasterStatus CompositorContext::ScopedFrame::Raster(
   if (post_preroll_result == PostPrerollResult::kSkipAndRetryFrame) {
     return RasterStatus::kSkipAndRetry;
   }
+
+  SkAutoCanvasRestore restore(canvas(), clip_rect.has_value());
```

Comment:
why do we need this?

Reply:
We don't if we always start with "fresh" canvas. Which seems to be the case right now for some reason, I'm not quite sure what has changed. I used to have the issue where `canvas()->clipRect(*clip_rect);` was making the rect smaller and smaller every frame until nothing was painted. Restoring the clip after rasterization fixed that of course.

It doesn't seem to be needed right now. Still I don't think removing the clip rect after rasterization is a bad thing.